xbmc/platform/android/MemUtils.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "utils/MemUtils.h"
+
+#include <stdlib.h>
+
+#undef ALIGN
+#define ALIGN(value, alignment) (((value)+(alignment-1))&~(alignment-1))
+
+namespace KODI
+{
+namespace MEMORY
+{
+
+// aligned memory allocation.
+// in order to do so - we alloc extra space and store the original allocation in it (so that we can free later on).
+// the returned address will be the nearest aligned address within the space allocated.
+void* AlignedMalloc(size_t s, size_t alignTo)
+{
+  if (CJNIBase::GetSDKVersion() >= 28)
+  {
+    return aligned_alloc(alignTo, s);
+  }
+  else
+  {
+    char *pFull = (char*)malloc(s + alignTo + sizeof(char *));
```

Comment:
use `posix_memalign`

Reply:
@lrusak I agree. `malloc` is spelled the same everywhere but should still be placed in platform code here because the implementations differ on every platform.